### PR TITLE
Erik o Eriks datetime.datetime

### PR DESCRIPTION
**Pull request by erikvader**

### CODE REVIEW

**Comment by theresehofmann** [comment]
The code is easy to follow and understand. 
Some test names like “testCreateArgs()” could be changed, it might be better to name it “testCreateDateTime”. 
You could have a test where you have another input type than int. 
It also seemed like you had some tests which you hadn’t finished yet, on the places it says todo (which you probably also are aware of).

File: python/common/org/python/stdlib/datetime/DateTime.java [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+		}
+
+		if (this.timeUnits[YEAR_INDEX] < MIN_YEAR || this.timeUnits[YEAR_INDEX] > MAX_YEAR) {
+			throw new org.python.exceptions.ValueError("year " + this.timeUnits[YEAR_INDEX] + "is out of range");
+		}
+
+		if (this.timeUnits[MONTH_INDEX] < 1 || this.timeUnits[MONTH_INDEX] > 12) {
+			throw new org.python.exceptions.ValueError("month " + this.timeUnits[MONTH_INDEX] + "is out of range");
+		}
+		if (this.timeUnits[DAY_INDEX] < 1 || this.timeUnits[DAY_INDEX] > 31) {
```
Comment (nystrand1) [inline python/common/org/python/stdlib/datetime/DateTime.java:85]: This will yield unexpected behaviour for months with less than 31 days. Also, with this february can have more than 28 days

Comment (nystrand1) [reply python/common/org/python/stdlib/datetime/DateTime.java:85]: Unexpected behaviour a.k.a invalid dates :D

File: python/common/org/python/stdlib/datetime/DateTime.java [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+	org.python.types.Int minut = org.python.types.Int.getInt(iMin);
+	org.python.types.Int secon = org.python.types.Int.getInt(iSec);
+	org.python.types.Int microsecon = org.python.types.Int.getInt(iMic); 
+
+	org.python.Object[] args = { yea, mont, da, hou, minut, secon, microsecon };
+	return new DateTime(args, Collections.emptyMap());
+
+	}
+
+	@org.python.Method(__doc__ = "Checks if current dateobject is less than dt")
```
Comment (nystrand1) [inline python/common/org/python/stdlib/datetime/DateTime.java:331]: Nicely done with the comparision-methods!

File: jtests/DateTimeTest.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        DateTime dt = new DateTime(args, kwargs);
+
+        assertEquals(dt.year, dt.__year__());
+        assertEquals(dt.month, dt.__month__());
+        assertEquals(dt.day, dt.__day__());
+        assertEquals(dt.hour, dt.__hour__());
+        assertEquals(dt.minute, dt.__minute__());
+        assertEquals(dt.second, dt.__second__());
+        assertEquals(dt.microsecond, dt.__microsecond__());
+    }
```
Comment (nystrand1) [inline jtests/DateTimeTest.java:242]: Add tests for date edge-cases like end of month etc.

**Comment by ErikKrantz** [comment]
> Some test names like “testCreateArgs()” could be changed, it might be better to name it “testCreateDateTime”.

As it is not simply a test for create DateTime we think we will keep it as it is. Since we want separate tests for args and kwargs. Perhaps testCreateDateTimeArgs would be more descriptive, or maybe too much? 

> You could have a test where you have another input type than int.

Nice catch! Added a test for floats as well! 

Thanks for the input!

File: jtests/DateTimeTest.java (hunk 3 above)
Comment (ErikKrantz) [reply jtests/DateTimeTest.java:242]: I do not quite follow you. The tests here are not used for edgecases/test of values. They are used to check whether the __minute__,__second__ .. methods are working as intended. :)

Comment (ErikKrantz) [reply jtests/DateTimeTest.java:242]: "__minute__","__second__" (apparently double underscore makes it bold)

Comment (ErikKrantz) [reply jtests/DateTimeTest.java:242]: \_\_minute\_\_ , \_\_second\_\_ **

File: python/common/org/python/stdlib/datetime/DateTime.java (hunk 2 above)
Comment (erikvader) [reply python/common/org/python/stdlib/datetime/DateTime.java:331]: ty

File: python/common/org/python/stdlib/datetime/DateTime.java (hunk 1 above)
Comment (erikvader) [reply python/common/org/python/stdlib/datetime/DateTime.java:85]: true, fixed now

File: jtests/DateTimeTest.java (hunk 3 above)
Comment (nystrand1) [reply jtests/DateTimeTest.java:242]: Sorry for being unclear, I meant that tests for those edge cases is needed but not necessarly in this particular place :D

Comment (ErikKrantz) [reply jtests/DateTimeTest.java:242]: Ah, alright! :D
